### PR TITLE
Start to updates the what's new / changelog.

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

**Comment by Carreau** [comment]
@ivanov this simplify a bit and prepare the release process. You expressed interest in trying it, let me know.

File: docs/source/whatsnew/index.rst [hunk 1]
```diff
@@ -12,6 +12,29 @@
 What's new in IPython
 =====================
 
+..
+    this will appear in the docs if we are nto releasing a versin (ie is
+    `_version_extra` in release.py is empty stringA
+
+.. only:: ipydev
+
+   Developpement version in-progress features:
```
Comment (minrk) [inline docs/source/whatsnew/index.rst]: Development

File: docs/source/whatsnew/version7.rst [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+~~~~~~
+
+This took more than a year in the making, and the code was rebased a number of
+time leading to commit authorship that may have been lost in the final
+Pull-Request. Huge thanks to many people for contribution, discussion, code,
+documentation, use-case: dalejung, danielballan, ellisonbg, fperez, gnestor,
+minrk, njsmith, pganssle, tacaswell, takluyver , vidartf ... And many other.
+
+
+Autoreload Improvment
```
Comment (minrk) [inline docs/source/whatsnew/version7.rst]: Improvement

**Review by minrk — APPROVED** [review]

**Comment by Carreau** [comment]
Typos fixed (and a couple of others as well).

**Comment by ivanov** [comment]
I built the docs from the [7.0.0b1](https://github.com/ipython/ipython/tree/7.0.0b1) tag and it looks like they still included the 'development' portion (which I did not remove because the release docs said I don't have to worry about that, that that section will be removed. Did I goof something up or do the docs need to be updated?

**Pull request merged**